src/prpy/planning/openrave.py

```diff
@@ -87,17 +89,17 @@ def _Plan(self, robot, goals, maxiter=500, continue_planner=False,
         params.SetGoalConfig(goals)
         params.SetExtraParameters(extraParams)
 
-        traj = openravepy.RaveCreateTrajectory(self.env, 'GenericTrajectory')
+        traj = openravepy.RaveCreateTrajectory(env, 'GenericTrajectory')
 
         try:
-            self.env.Lock()
+            env.Lock()
 
             # Plan.
             if (not continue_planner) or not self.setup:
-                self.planner.InitPlan(robot, params)
+                planner.InitPlan(robot, params)
                 self.setup = True
 
-            status = self.planner.PlanPath(traj, releasegil=True)
+            status = planner.PlanPath(traj, releasegil=True)
```

Comment:
This should be wrapped in a `RobotStateSaver` that saves `robot`'s `LinkTransformation`s.